SMLHelper/Utility/EnumCacheManager.cs

```diff
@@ -36,14 +38,55 @@ internal EnumCacheManager(string enumTypeName, int startingIndex, IEnumerable<in
 
         #region Caching
 
-        private string GetCachePath()
+        private string GetCacheDirectoryPath()
         {
             var saveDir = @"./QMods/Modding Helper/" + $"{enumTypeName}Cache";
 
             if (!Directory.Exists(saveDir))
                 Directory.CreateDirectory(saveDir);
 
-            return Path.Combine(saveDir, $"{enumTypeName}Cache.txt");
+            return saveDir;
+        }
+
+        private string GetCachePath()
+        {
+            return Path.Combine(GetCacheDirectoryPath(), $"{enumTypeName}Cache.txt");
+        }
+
+        private string GetBanListPath()
+        {
+            return Path.Combine(GetCacheDirectoryPath(), "Banlist.txt");
+        }
```

Comment:
This means that users installing need to manually edit the ban list, because it can't be overridden, or else other banned indices might be lost

Reply:
I think it would be better if you could have a banned indices folder where people could add their own text files, instead of editing the same one

Reply:
From what I'm seeing, this external file loads indexes into the `bannedIndices` collection.
This collection can also be instantiated through the constructor `EnumCacheManager(string enumTypeName, int startingIndex, IEnumerable<int> bannedIndices)` to provide an initial hard coded list.

So the external indexes get combined with the internal ones.

Reply:
Yes, this would be more flexible since having a folder of multiple files means a simple "drag and drop" action for the end user rather than opening a file, copy/pasting text, and saving it.

Reply:
So, basically something like this?
Modding Helper/Banned Ids/TestMod.txt
Modding Helper/Banned Ids/TestMod2.txt

And TestMod.txt can be something like this:
```
1111:TechType
10:CraftTreeType
```

Reply:
yes

Reply:
But keep the folder name without spaces (like the other ones)